### DISCUSSION TITLE
Backup and restore with Velero suggestion

### DISCUSSION THREAD

**Discussion by veerendra2** [strimzi/strimzi-kafka-operator#5894] [Q&A] [answered]
I know this is not Velero project. I just need some info from Strimzi side. 
### Background Info
* I recently started using Strimzi and choose Velero as disaster recovery tool. While performing backup and restore it is consuming lot(significant amount) of CPU and MEM due to backup of entire namespace/deployments,etc. Sometime restore is failing due trying to restore unnecessary resources(not sure why)
* I want to cut down time and resource consumption. 
### Need info
Is it ok if I redeploy entire strimzi kafka cluster including cluster operator with old PVCs? 
For example;
I have kafka cluster running 
```
$ oc get pods
NAME                                          READY     STATUS    RESTARTS   AGE
sirius-dev-entity-operator-5d6dbddf46-67c52   3/3       Running   0          20h
sirius-dev-kafka-0                            1/1       Running   0          20h
sirius-dev-kafka-1                            1/1       Running   0          20h
sirius-dev-kafka-2                            1/1       Running   0          20h
sirius-dev-kafka-exporter-695c4b88cb-6qpqm    1/1       Running   0          20h
sirius-dev-zookeeper-0                        1/1       Running   0          20h
sirius-dev-zookeeper-1                        1/1       Running   0          20h
sirius-dev-zookeeper-2                        1/1       Running   0          20h
strimzi-cluster-operator-5c8d5cf966-nfz55     1/1       Running   8          21h
```
And PVC
```

$ oc get pvc
NAME                          STATUS    VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
data-0-sirius-dev-kafka-0     Bound     pvc-63c4719a-3e3a-11ec-9048-000d3ab3f21d   256Gi      RWO            generic        21h
data-0-sirius-dev-kafka-1     Bound     pvc-63c4b98b-3e3a-11ec-9048-000d3ab3f21d   256Gi      RWO            generic        21h
data-0-sirius-dev-kafka-2     Bound     pvc-63c4e212-3e3a-11ec-9048-000d3ab3f21d   256Gi      RWO            generic        21h
data-sirius-dev-zookeeper-0   Bound     pvc-30d39923-3e3a-11ec-9048-000d3ab3f21d   64Gi       RWO            generic        21h
data-sirius-dev-zookeeper-1   Bound     pvc-30d353bd-3e3a-11ec-9048-000d3ab3f21d   64Gi       RWO            generic        21h
data-sirius-dev-zookeeper-2   Bound     pvc-30dc276e-3e3a-11ec-9048-000d3ab3f21d   64Gi       RWO            generic        21h
```
* Delete everything in namespace except PVC.
* Deployment kafka cluster again with same configures
* The new deployment picks up the existing PVC and start using it as a disk

Not sure how strimzi handles internally I do like above

What I'm trying to achieve is simulating disaster recovery

Thanks

**Comment by scholzj** [comment, marked as answer]
You should check this guide which is about recovering the cluster when all you have is the storage: https://strimzi.io/docs/operators/latest/full/using.html#cluster-recovery-str

In your case, if you have the PVCs, you obviously do not need to recreate them. But the rest should apply to you I guess.

**Reply by veerendra2** [reply]
Thanks for pointing out docs and quick response
